directv: extended discovery via REST api, bug fix

sielicki opened the pull request:
## Description:

See commit messages for better details.

DirecTV RVU clients are now detected and configured if the discovery component is used.

**Related issue (if applicable):** 
- Probably can close home-assistant/netdisco#42 as a wont-fix or not-a-bug
- Fixes #9003

**Pull request in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io) with documentation (if applicable):** home-assistant/home-assistant.github.io#3128

## Checklist:

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io)

If the code communicates with devices, web services, or third-party tools:
 - [x] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
- [x] New dependencies are only imported inside functions that use them ([example][ex-import]).
- [x] New dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
- [x] New dependencies have been added to the `REQUIREMENTS` variable ([example][ex-requir]).
- [x] New files were added to `.coveragerc`.

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L14
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L54

sielicki committed:
fix not providing device for discovered directvs

This fixes a bug introduced at 6884965c80

Discovered directv boxes would not be instantiated with a DEVICE
parameter.

Signed-off-by: Nicholas Sielicki <sielicki@yandex.com>

sielicki committed:
directv: add discovery of RVU clients

If discovery is used with directv, also try to further discover and
configure RVU client set-top boxes by requesting information from a REST
service running on the main directv box/RVU-server.

This commit also disables discovery if any directv configuration is
supplied by the user.

Signed-off-by: Nicholas Sielicki <sielicki@yandex.com>

pvizeli commented on homeassistant/components/media_player/directv.py:
```diff
@@ -25,7 +25,7 @@
     SUPPORT_PREVIOUS_TRACK | SUPPORT_PLAY
 
-KNOWN_HOSTS = []
+KNOWN_DEVICES = []
```
Globals like this are not allow anymore. Use `hass.data`

sielicki committed:
components/media_player/directv.py: use hass.data

Use hass.data instead of a global to remember state.

Signed-off-by: Nicholas Sielicki <sielicki@yandex.com>

sielicki commented:
Please review this hass.data commit with an extra dose of skepticism, I'm just cargo-culting from other modules.

sielicki commented:
Ignore the failed CI build, this is due to https://github.com/home-assistant/home-assistant/pull/8838

andrey-git commented:
Any updates on this PR

pvizeli commented on homeassistant/components/media_player/directv.py:
```diff
@@ -37,32 +37,46 @@
+        name = 'DirecTV_' + discovery_info.get('serial', '')
+
+        # attempt to discover additional RVU units
+        import requests
```
Import that on top, it is a core requerement

pvizeli reviewed (changes requested):
Small comment and we can merg it

sielicki committed:
unconditionally import requests in directv.py

Requests is a core requirement, so we're okay to import at the top of
the file rather than conditionally / in a function.

Signed-off-by: Nicholas Sielicki <sielicki@yandex.com>
